### ISSUE TITLE
bcl2fastq link error with libxslt-1.1.34

### ISSUE THREAD

**Issue by verdurin** [easybuilders/easybuild-easyconfigs#13981]
Was trying to build `SpaceRanger-1.3.0` and saw this link error with `bcl2fastq`, one of the dependencies:
```
[100%] Linking CXX executable bcl2fastq
cd /dev/shm/bcl2fastq2/2.20.0/GCC-10.3.0/easybuild_obj/cxx/bin && /gpfs3/apps/eb/2020b/skylake/software/CMake/3.20.1-GCCcore-10.3.0/bin/cmake -E cmake_link_script CMakeFiles/bcl2fastq.dir/link.txt --verbose=1
/apps/eb/2020b/skylake/software/GCCcore/10.3.0/bin/g++  -O2 -ftree-vectorize -march=native -fno-math-errno -fPIC -std=c++11 -fopenmp -msse2 -Wall -Wextra -Wunused -Wno-long-long -Wsign-compare -Wpointer-arith -Wno-unused-parameter  -DNDEBUG -L/apps/eb/2020b/skylake/software/zlib/1.2.11-GCCcore-10.3.0/lib64 -L/apps/eb/2020b/skylake/software/zlib/1.2.11-GCCcore-10.3.0/lib -L/apps/eb/2020b/skylake/software/libxslt/1.1.34-GCCcore-10.3.0/lib64 -L/apps/eb/2020b/skylake/software/libxslt/1.1.34-GCCcore-10.3.0/lib -L/apps/eb/2020b/skylake/software/libxml2/2.9.10-GCCcore-10.3.0/lib64 -L/apps/eb/2020b/skylake/software/libxml2/2.9.10-GCCcore-10.3.0/lib -L/apps/eb/2020b/skylake/software/GCCcore/10.3.0/lib64 -L/apps/eb/2020b/skylake/software/GCCcore/10.3.0/lib -rdynamic CMakeFiles/bcl2fastq.dir/bcl2fastq.cpp.o -o bcl2fastq  ../lib/conversion/libbcl2fastq_conversion.a ../lib/stats/libbcl2fastq_stats.a ../lib/layout/libbcl2fastq_layout.a ../lib/data/libbcl2fastq_data.a ../lib/io/libbcl2fastq_io.a ../lib/config/libbcl2fastq_config.a ../lib/common/libbcl2fastq_common.a ../../bootstrap/lib/libboost_chrono.a ../../bootstrap/lib/libboost_date_time.a ../../bootstrap/lib/libboost_filesystem.a ../../bootstrap/lib/libboost_iostreams.a ../../bootstrap/lib/libboost_program_options.a ../../bootstrap/lib/libboost_regex.a ../../bootstrap/lib/libboost_serialization.a ../../bootstrap/lib/libboost_system.a ../../bootstrap/lib/libboost_timer.a ../../bootstrap/lib/libboost_thread.a -lpthread -Wl,-Bstatic -lexslt -Wl,-Bdynamic -lxslt -lxml2 -lz -lrt -lpthread
/apps/eb/2020b/skylake/software/libxslt/1.1.34-GCCcore-10.3.0/lib64/libexslt.a(crypto.o):crypto.c:function exsltCryptoMd5Function: error: undefined reference to 'gcry_md_hash_buffer'
/apps/eb/2020b/skylake/software/libxslt/1.1.34-GCCcore-10.3.0/lib64/libexslt.a(crypto.o):crypto.c:function exsltCryptoMd5Function: error: undefined reference to 'gcry_check_version'
/apps/eb/2020b/skylake/software/libxslt/1.1.34-GCCcore-10.3.0/lib64/libexslt.a(crypto.o):crypto.c:function exsltCryptoRc4EncryptFunction: error: undefined reference to 'gcry_cipher_open'
/apps/eb/2020b/skylake/software/libxslt/1.1.34-GCCcore-10.3.0/lib64/libexslt.a(crypto.o):crypto.c:function exsltCryptoRc4EncryptFunction: error: undefined reference to 'gcry_cipher_setkey'
/apps/eb/2020b/skylake/software/libxslt/1.1.34-GCCcore-10.3.0/lib64/libexslt.a(crypto.o):crypto.c:function exsltCryptoRc4EncryptFunction: error: undefined reference to 'gcry_cipher_encrypt'
/apps/eb/2020b/skylake/software/libxslt/1.1.34-GCCcore-10.3.0/lib64/libexslt.a(crypto.o):crypto.c:function exsltCryptoRc4EncryptFunction: error: undefined reference to 'gcry_cipher_close'
/apps/eb/2020b/skylake/software/libxslt/1.1.34-GCCcore-10.3.0/lib64/libexslt.a(crypto.o):crypto.c:function exsltCryptoRc4EncryptFunction: error: undefined reference to 'gcry_strerror'
/apps/eb/2020b/skylake/software/libxslt/1.1.34-GCCcore-10.3.0/lib64/libexslt.a(crypto.o):crypto.c:function exsltCryptoRc4EncryptFunction: error: undefined reference to 'gcry_strerror'
/apps/eb/2020b/skylake/software/libxslt/1.1.34-GCCcore-10.3.0/lib64/libexslt.a(crypto.o):crypto.c:function exsltCryptoRc4EncryptFunction: error: undefined reference to 'gcry_strerror'
/apps/eb/2020b/skylake/software/libxslt/1.1.34-GCCcore-10.3.0/lib64/libexslt.a(crypto.o):crypto.c:function exsltCryptoRc4EncryptFunction: error: undefined reference to 'gcry_check_version'
/apps/eb/2020b/skylake/software/libxslt/1.1.34-GCCcore-10.3.0/lib64/libexslt.a(crypto.o):crypto.c:function exsltCryptoRc4DecryptFunction: error: undefined reference to 'gcry_cipher_open'
/apps/eb/2020b/skylake/software/libxslt/1.1.34-GCCcore-10.3.0/lib64/libexslt.a(crypto.o):crypto.c:function exsltCryptoRc4DecryptFunction: error: undefined reference to 'gcry_cipher_setkey'
/apps/eb/2020b/skylake/software/libxslt/1.1.34-GCCcore-10.3.0/lib64/libexslt.a(crypto.o):crypto.c:function exsltCryptoRc4DecryptFunction: error: undefined reference to 'gcry_cipher_decrypt'
/apps/eb/2020b/skylake/software/libxslt/1.1.34-GCCcore-10.3.0/lib64/libexslt.a(crypto.o):crypto.c:function exsltCryptoRc4DecryptFunction: error: undefined reference to 'gcry_cipher_close'
/apps/eb/2020b/skylake/software/libxslt/1.1.34-GCCcore-10.3.0/lib64/libexslt.a(crypto.o):crypto.c:function exsltCryptoRc4DecryptFunction: error: undefined reference to 'gcry_strerror'
/apps/eb/2020b/skylake/software/libxslt/1.1.34-GCCcore-10.3.0/lib64/libexslt.a(crypto.o):crypto.c:function exsltCryptoRc4DecryptFunction: error: undefined reference to 'gcry_check_version'
/apps/eb/2020b/skylake/software/libxslt/1.1.34-GCCcore-10.3.0/lib64/libexslt.a(crypto.o):crypto.c:function exsltCryptoMd4Function: error: undefined reference to 'gcry_md_hash_buffer'
/apps/eb/2020b/skylake/software/libxslt/1.1.34-GCCcore-10.3.0/lib64/libexslt.a(crypto.o):crypto.c:function exsltCryptoMd4Function: error: undefined reference to 'gcry_check_version'
/apps/eb/2020b/skylake/software/libxslt/1.1.34-GCCcore-10.3.0/lib64/libexslt.a(crypto.o):crypto.c:function exsltCryptoSha1Function: error: undefined reference to 'gcry_md_hash_buffer'
collect2: error: ld returned 1 exit status
make[2]: *** [cxx/bin/bcl2fastq] Error 1
make[2]: Leaving directory `/dev/shm/bcl2fastq2/2.20.0/GCC-10.3.0/easybuild_obj'
make[1]: *** [cxx/bin/CMakeFiles/bcl2fastq.dir/all] Error 2
make[1]: Leaving directory `/dev/shm/bcl2fastq2/2.20.0/GCC-10.3.0/easybuild_obj'
make: *** [all] Error 2
```

**Comment by verdurin** [comment]
This appears to be caused by building `libxslt` on a node that has `libgcrypt-devel` installed, which means that the crypto extensions are included.

**Comment by SethosII** [comment]
@verdurin I also hit this bug while building `bcl2fastq`. Have you solved this?

**Comment by akesandgren** [comment]
Working on a fix...

**Comment by akesandgren** [comment]
Fixed in #14756 which makes sure libxslt doesn't pick up on external libgcrypt
